### PR TITLE
Fix api-reference shortcode for localized pages.

### LINKED ISSUE #34178: `api-reference` shortcode is not working in localized pages
**This is a Bug Report**

**Problem:**
`api-reference` shortcode is not working in localized pages.
@s-kawamura-w664 found this in the PR #32658.

- en: https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#links-to-api-reference
- ja: https://deploy-preview-32658--kubernetes-io-main-staging.netlify.app/ja/docs/contribute/style/hugo-shortcodes/#api%E3%83%AA%E3%83%95%E3%82%A1%E3%83%AC%E3%83%B3%E3%82%B9%E3%81%B8%E3%81%AE%E3%83%AA%E3%83%B3%E3%82%AF
- https://github.com/kubernetes/website/pull/32658#pullrequestreview-999408661
- https://github.com/kubernetes/website/pull/32658#discussion_r892130849
- https://github.com/kubernetes/website/pull/32658#discussion_r892131010
- https://kubernetes.io/ko/docs/concepts/workloads/controllers/cron-jobs/#%EB%8B%A4%EC%9D%8C-%EB%82%B4%EC%9A%A9
- https://kubernetes.io/zh/docs/concepts/workloads/controllers/daemonset/#what-s-next
- ...

**Proposed Solution:**

**File to Update:**
`layouts/shortcodes/api-reference.html`

### PR DESCRIPTION
**Pull request by s-kawamura-w664**
Fixes #34178

Fix `api-reference` shortcode not working on localized pages.

### CODE REVIEW

**Comment by jihoon-seo** [comment]
Previews:
- if the localized version of api reference for current lang does not exist,
  then `api-reference` shortcode generates a link that points to the English version of api reference.
  - [See this](https://deploy-preview-34272--kubernetes-io-main-staging.netlify.app/ko/docs/concepts/workloads/controllers/cron-jobs/#%EB%8B%A4%EC%9D%8C-%EB%82%B4%EC%9A%A9:~:text=API%EC%9D%98%20%EC%9D%BC%EB%B6%80%EC%9D%B4%EB%8B%A4.-,CronJob,-%EC%98%A4%EB%B8%8C%EC%A0%9D%ED%8A%B8%20%EC%A0%95%EC%9D%98%EB%A5%BC%20%EC%9D%BD%EA%B3%A0)
  - [See this](https://deploy-preview-34272--kubernetes-io-main-staging.netlify.app/zh-cn/docs/concepts/workloads/controllers/daemonset/#%E6%8E%A5%E4%B8%8B%E6%9D%A5:~:text=%E7%BA%A7%E8%B5%84%E6%BA%90%E3%80%82%E9%98%85%E8%AF%BB-,DaemonSet,-%E5%AF%B9%E8%B1%A1%E5%AE%9A%E4%B9%89%E7%90%86%E8%A7%A3)
- if the localized version of api reference for current lang exists (e.g. [Chinese version](https://github.com/kubernetes/website/blob/main/content/zh-cn/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1.md)),
  then api-reference shortcode generates a link that points to the localized version of api reference.
  - [See this](https://deploy-preview-34272--kubernetes-io-main-staging.netlify.app/zh-cn/docs/tasks/configure-pod-container/pull-image-private-registry/#:~:text=%E6%88%96%E9%98%85%E8%AF%BB-,Secret,-%E7%9A%84%20API%20%E5%8F%82%E8%80%83)

This is a great fix! Thank you @s-kawamura-w664 !
/lgtm

**Comment by jihoon-seo** [comment]
Action item:
(After this PR gets reviewed and merged,) backport this to

- `release-1.22`
- `release-1.23`

**Comment by s-kawamura-w664** [comment]
/assign @sftim @tengqm 

Could you please confirm this PR.

**Review by sftim — APPROVED** [review]
/area web-development

Thanks. Happy to approve this.

**Comment by s-kawamura-w664** [comment]
@sftim 
Thank you for your advice on assigning approvers.
I had assigned from the past history of the modified file.
I will assign as you advise in the future.

**Pull request merged**
